data/card_definitions.json

```diff
@@ -1,4 +1,310 @@
+			"effects": [
+				{
+					"timing": "now",
+					"effect_type": "choice",
+					"choice": [
+						{ "effect_type": "spend_life", "amount": 1, "and": { "effect_type": "draw", "amount": 1 } },
+						{ "effect_type": "spend_life", "amount": 2, "and": { "effect_type": "draw", "amount": 2 } },
+						{ "effect_type": "spend_life", "amount": 3, "and": { "effect_type": "draw", "amount": 3 } },
+						{ "effect_type": "spend_life", "amount": 4, "and": { "effect_type": "draw", "amount": 4 } },
+						{ "effect_type": "spend_life", "amount": 5, "and": { "effect_type": "draw", "amount": 5 } },
+						{ "effect_type": "pass" }
+					]
+				},
+				{
+					"timing": "during_strike",
+					"effect_type": "powerup",
+					"amount": 1	
+				}
+			]
+		}		
+	},
+	{
+		"id": "akuma_goshoryuken",
+		"type": "special",
+		"display_name": "Goshoryuken",
+		"force_cost": 0,
+		"gauge_cost": 0,
+		"range_min": 1,
+		"range_max": 1,
+		"speed": 7,
+		"power": 3,
+		"armor": 0,
+		"guard": 0,
+		"effects": [
+			{
+				"timing": "before",
+				"effect_type": "close",
+				"amount": 1
+			},
+			{
+				"timing": "hit",
+				"effect_type": "push",
+				"amount": 2
+			}
+		],	
+		"boost": {
+			"boost_type": "immediate",
+			"force_cost": 1,
+			"cancel_cost": -1,
+			"display_name": "Ashura Senku",
+			"effects": [
+				{
+					"timing": "immediate",
+					"effect_type": "choice",
+					"choice": [
+						{ "effect_type": "add_to_gauge_boost_play_cleanup" },
+						{ "effect_type": "advance", "amount": 1 },
+						{ "effect_type": "retreat", "amount": 1 }
+					]
+				},
+				{
+					"timing": "immediate",
+					"effect_type": "bonus_action"
+				}
+			]
+		}		
+	},
+	{
+		"id": "akuma_hyakkishu",
+		"type": "special",
+		"display_name": "Hyakkishu",
+		"force_cost": 0,
+		"gauge_cost": 0,
+		"range_min": 1,
+		"range_max": 1,
+		"speed": 4,
+		"power": 4,
+		"armor": 0,
+		"guard": 0,
+		"effects": [
+			{
+				"timing": "during_strike",
+				"condition": "is_critical",
+				"effect_type": "ignore_armor"
+			},
+			{
+				"timing": "before",
+				"effect_type": "close",
+				"amount": 1
+			},
+			{
+				"timing": "after",
+				"effect_type": "dodge_at_range",
+				"range_min": 1,
+				"range_max": 1
+			}
+		],
+		"boost": {
+			"boost_type": "continuous",
+			"force_cost": 0,
+			"cancel_cost": -1,
+			"display_name": "Fakeout",
+			"effects": [
+				{
+					"timing": "now",
+					"effect_type": "strike_opponent_sets_first"	
+				},
+				{
+					"timing": "after",
+					"effect_type": "add_boost_to_gauge_on_strike_cleanup",
+					"card_name": "Hyakkishu"
+				}
+			]
+		}		
+	},
+	{
+		"id": "akuma_tatsumakizankukyaku",
+		"type": "special",
+		"display_name": "Tatsumaki Zankukyaku",
+		"force_cost": 0,
+		"gauge_cost": 0,
+		"range_min": 1,
+		"range_max": 2,
+		"speed": 4,
+		"power": 4,
+		"armor": 0,
+		"guard": 0,
+		"effects": [
+			{
+				"timing": "before",
+				"effect_type": "advance",
+				"amount": 2,
+				"bonus_effect": {
```

Comment:
this part is supposed to be a hit effect.
maybe use set_strike_x as the advanced through part and set it to one, and then have a hit effect to discard the set x value?

Reply:
ryu's tatsu is coded the same way but works in the way you think by only gaining advantage when hitting the opponent, when i tested by using hit effect it errored out.

Reply:
that's probably wrong too then

Reply:
oh yeah tatsu probably is wrong; i'll look into making this kind of thing easier to write later

Reply:
ok probably a good change since ranged dodges is a thing and don't want ryu to be gaining advantage for free